### PR TITLE
added xs_os.py and its test, reformatted using black, checked and ran using pytest and python.

### PR DESCRIPTION
**Pull request by JosephDettmann**
added xs_os.py and its test, reformatted using black, checked and ran using pytest and python. accidentally created this branch from complementary_dna, changed the file to match the way it is in main so that the file isn't altered.

### CODE REVIEW

File: src/xs_os.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+def xs_os(s):
+    s = s.lower()
+    return s.count("x") == s.count("o")
```
Comment (W-Ely) [inline src/xs_os.py:3]: Avoid single character variable names.

File: src/comp_dna.py [hunk 2]
```diff
@@ -1,15 +1,6 @@
...
-            comp_dna = comp_dna + "G"
-        elif char == "T":
-            comp_dna = comp_dna + "A"
-        elif char == "G":
-            comp_dna = comp_dna + "C"
-        elif char == "A":
-            comp_dna = comp_dna + "T"
-    return comp_dna
+    trans_map = {"A": "T", "T": "A", "C": "G", "G": "C"}
+    return "".join([trans_map[x] for x in dna])
```
Comment (W-Ely) [inline src/comp_dna.py:3]: Checkout to main and pull the latest changes.  Then checkout back to this branch.  Then run "git merge main" to merge the changes from main into this branch.  That way this diff doesn't show up.

Comment (JosephDettmann) [reply src/comp_dna.py:3]: okey dokey, thank you!

Comment (W-Ely) [reply src/comp_dna.py:3]: Also once the merge has happened locally, the branch still needs to be pushed back up to GitHub

**Pull request merged**